src/neo/Helper.cs

```diff
@@ -310,5 +310,88 @@ internal static BigInteger WeightedAverage<T>(this IEnumerable<T> source, Func<T
                 yield return resultSelector(item, weight);
             }
         }
+
+        public static byte[] Concat(this byte[] a, byte[] b)
+        {
+            var result = new byte[a.Length + b.Length];
+            a.CopyTo(result, 0);
+            b.CopyTo(result, a.Length);
+            return result;
+        }
+
+        public static byte[] CommonPrefix(this byte[] a, byte[] b)
+        {
+            var prefix = Array.Empty<byte>();
+            var minLen = a.Length <= b.Length ? a.Length : b.Length;
+
+            if (a.Length != 0 && b.Length != 0)
+            {
+                for (int i = 0; i < minLen; i++)
+                {
+                    if (a[i] != b[i]) break;
+                    prefix = prefix.Add(a[i]);
+                }
+            }
+            return prefix;
+        }
+
+        public static bool Equal(this byte[] a, byte[] b)
+        {
+            if (a is null || b is null) return false;
+            var len = a.Length;
+            if (len != b.Length) return false;
+            unsafe
```

Comment:
You've used the `unsafe` block to compare the byte array in bunches of 8 bytes at a time, that's clever, have you compared the performance of that with the simple approach?

Have you considered the `SequenceEqual`?
https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.sequenceequal?view=netcore-3.1

Reply:
I will do the test to compare the performance.

Reply:
`SequenceEqual` will take more time.